Source/Core/DotSpatial.Controls/MapLabelLayer.cs

```diff
@@ -667,9 +667,14 @@ private static float GetAngleToRotate(ILabelSymbolizer symb, IFeature feature, G
                 {
                     var ls = GetSegment(ls1, symb);
                     if (ls == null) return 0;
+
                     if (symb.LineOrientation == LineOrientation.Parallel)
-                        return ToSingle(-ls.Angle);
-                    return ToSingle(-ls.Angle - 90);
+                        //NTS value for Angle is in radians so must convert to degrees
+                        return ToSingle(-ls.Angle * (180.0 / Math.PI));
+
+                    //convert Angle to degrees and then adjust value by +90� to make it perpendicular.
+                    //an adjusment value of -90� would tend to make the label "upside down" for most projections.
+                    return ToSingle((-ls.Angle * (180.0 / Math.PI)) + 90.0);
```

Comment:
Replace your code with

```
 if (symb.LineOrientation == LineOrientation.Parallel)
                    {
                        return ToSingle(-GetDisplayAngle(ls));
                    }

                    return ToSingle(-GetDisplayAngle(ls) - 90);
```

Reply:
add the following method to the file:

```
/// <summary>
        /// Gets the angle that will be used to draw line based labels so they won't be on their head.
        /// </summary>
        private static double GetDisplayAngle(LineSegment ls)
        {
            double deltaX = ls.P1.X - ls.P0.X;
            double deltaY = ls.P1.Y - ls.P0.Y;
            double length = Math.Sqrt(deltaX * deltaX + deltaY * deltaY);
            double angleRAD = Math.Asin(Math.Abs(ls.P1.Y - ls.P0.Y) / length);
            
            if (angleRAD != ls.Angle)
            {

                Debug.WriteLine(angleRAD + "!=" + ls.Angle);
            }

            double angle = (angleRAD * 180) / Math.PI;

            if (((ls.P0.X < ls.P1.X) && (ls.P0.Y > ls.P1.Y)) || ((ls.P0.X > ls.P1.X) && (ls.P0.Y < ls.P1.Y)))
            {
                angle = 360 - angle;
            }

            return angle;
        }
```

Reply:
@sindizzy I had a look at your code and didn't like the fact that your east lines had labels that were on their head. Please replace it with the code I posted. This will give us labels that ain't on their head.

Reply:
When everythings is alright please remove the test files. The TestData folder should only contain files that are used for the automatic tests in the DotSpatial.Controls.Tests classes.